numpy/core/src/umath/loops.c.src

```diff
@@ -845,6 +846,109 @@ NPY_NO_EXPORT NPY_GCC_OPT_3 void
 
 /**end repeat1**/
 
+/*
+ * Arithmetic bit shift operations.
+ *
+ * Intel hardware masks bit shift values, so large shifts wrap around
+ * and can produce surprising results. The special handling ensures that
+ * behavior is independent of compiler or hardware.
+ * TODO: We could implement consistent behavior for negative shifts,
+ *       which is undefined in C.
+ */
+
+#define LEFT_SHIFT_OP                               \
+    do {                                            \
+        if (NPY_LIKELY(in2 < sizeof(@type@) * 8)) { \
```

Comment:
I think there are number of bits macros defined somewhere... yep, in `npy_common.h`. The have the form `NPY_BITSOF_LONGLONG` etc.

Reply:
Macro removed in #13739. Kept the if statements because I think they're clearer. Did not use `NPY_BITS_OF_LONG` and friends because they don't exist for `NPY_BITS_OF_ULONG` etc.